model/oauth/client.go

```diff
@@ -764,9 +764,9 @@ func validToken(i *instance.Instance, audience, token string) (permission.Claims
 		return claims, false
 	}
 	// Note: the refresh and registration tokens don't expire, no need to check its issue date
-	if claims.Audience != audience {
+	if len(claims.Audience) != 1 || claims.Audience[0] != audience {
```

Comment:
Adding an `Audience()` method to the `Claims` struct would help here. Something like this:

```go
func (c *Claims) Audience() String {
  if len(c.Audience) > 0 {
    return c.Audience[0]
  }
  return ""
}
```

Reply:
This code does not compile

Reply:
The returned type needs to be `string`, not `String` and there might be other syntax errors but do you get my point?

Reply:
We could use another name, like `func (c *Claims) AudienceString() string {}`

Reply:
@taratatach what do you think?

Reply:
Sorry, I missed your answers. Your proposition is totally fine by me :).